### PR TITLE
Fix Callsc Function Naming & Add From Address

### PR DESCRIPTION
**Pull request by pivilartisant**
Screenshot 2023-08-18 at 17 47 57

### CODE REVIEW

File: internal/handler/wallet/sign.go [hunk 1]
```diff
@@ -53,6 +56,20 @@ func (s *walletSign) Handle(params operations.SignParams) middleware.Responder {
 		return resp
 	}
 
+	// Fetch the account information for the wallet using the massaClient
+	infos, err := s.massaClient.GetAccountsInfos([]wallet.Wallet{*wlt})
+	if err != nil {
+		// Handle the error and return an internal server error response
+		errorMsg := fmt.Sprintf("Failed to fetch wallett infos: %s", err.Error())
```
Comment (Thykof) [inline internal/handler/wallet/sign.go:63]:
```suggestion
		errorMsg := fmt.Sprintf("Failed to fetch wallett infos: %w", err)
```

File: internal/handler/wallet/sign.go [hunk 2]
```diff
@@ -53,6 +56,20 @@ func (s *walletSign) Handle(params operations.SignParams) middleware.Responder {
 		return resp
 	}
 
+	// Fetch the account information for the wallet using the massaClient
```
Comment (Thykof) [inline internal/handler/wallet/sign.go:59]: you don't need massa client to get the address of the wallet.

**Review by Thykof — CHANGES_REQUESTED** [review]

File: wails-frontend/src/pages/passwordPrompt.tsx [hunk 3]
```diff
@@ -42,7 +42,8 @@ interface PromptRequestCallSCData {
   GasLimit: number;
   Coins: number;
   Address: string;
-  CalledFunction: string;
+  Function: string;
+  FromAddress: string;
```
[the remark this answers is not in the document]
Comment (B-Naoufal) [reply wails-frontend/src/pages/passwordPrompt.tsx:46]: it's the address of the wallet

[the remark this answers is not in the document]
Comment (pivilartisant) [reply wails-frontend/src/pages/passwordPrompt.tsx:46]: Screenshot 2023-08-18 at 17 44 58

Indeed, in design we expect two addresses

[the remark this answers is not in the document]
Comment (Thykof) [reply wails-frontend/src/pages/passwordPrompt.tsx:46]: the wallet that will sign the operation, how do you call it in core API or blockfeeder? Could be great to have the same name. Here it makes me thing of a transfer, but if I understand well it's not always a XMA transafer rigth?

[the remark this answers is not in the document]
Comment (B-Naoufal) [reply wails-frontend/src/pages/passwordPrompt.tsx:46]: well basically you authorize your wallet (from address)  to spend coins by signing the transaction  (it implicitly transfer some coins to the smart contract address)

File: internal/handler/wallet/sign.go (hunk 2 above)
Comment (B-Naoufal) [reply internal/handler/wallet/sign.go]: done

**Review by B-Naoufal — APPROVED** [review]

**Review by Thykof — APPROVED** [review]

**Review by Thykof — CHANGES_REQUESTED** [review]
please fix this https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&types=BUG&pullRequest=616&id=massalabs_thyra-plugin-wallet&open=AYoJSgWbvlejzMTBs9Nj

**Comment by pivilartisant** [comment]
Screenshot 2023-08-18 at 17 55 40

Recent changes to address have broke FromAddress

**Review by Thykof — APPROVED** [review]

**Comment by longgamma-massa** [comment]
Q/A needed?

**Comment by Thykof** [comment]
screenshot of the web on chain endpoint: QA OK on my side

Screenshot 2023-08-21 at 17 12 47

**Pull request merged**